### DISCUSSION TITLE
How to build a list of osm tags recognized by Photon.

### DISCUSSION THREAD

**Discussion by radim-asamm** [komoot/photon#710] [Q&A] [answered]
Hi all. What I need to do: Using `osm_key` and `osm_value` fields in the photon response, I need to build some generic categorization of results. Resolution of icon is one motivation, but there are more. It would be very useful to build a list of all possible (recognized) key:value combinations and then map these tags to a tree of categories that I already have for some other purpose. I found [import-full.style](https://github.com/osm-search/Nominatim/blob/master/settings/import-full.style). Is parsing this file the best way how to build such a list? Thanks, any advice is appreciated.

**Comment by lonvia** [comment, marked as answer]
The values for `osm_key` and `osm_values` are indeed taken directly from whatever Nominatim imports. So, yes, parsing the style file is the best way to get the list. 

 If you use the data dumps we provide, then the database was built using a slightly adapted variant of `import-full.style`.
